src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1426,14 +1434,25 @@ private IEnumerable<ITypeSymbol> InferTypeInNameColon(NameColonSyntax nameColon,
                 return SpecializedCollections.EmptyEnumerable<ITypeSymbol>();
             }
 
-            private IEnumerable<ITypeSymbol> InferTypeInMemberAccessExpression(MemberAccessExpressionSyntax expression)
+            private IEnumerable<ITypeSymbol> InferTypeInMemberAccessExpression(MemberAccessExpressionSyntax expression, SyntaxToken previousToken = default(SyntaxToken))
             {
                 var awaitExpression = expression.GetAncestor<AwaitExpressionSyntax>();
                 if (awaitExpression != null)
                 {
                     return InferTypes(awaitExpression.Expression);
                 }
 
+                if (previousToken != default(SyntaxToken) && previousToken == expression.OperatorToken)
+                {
+                    if (expression.Parent is MemberAccessExpressionSyntax)
```

Comment:
Ok.  Here's what this method should look like:

```c#


            private IEnumerable<ITypeSymbol> InferTypeInMemberAccessExpression(
                MemberAccessExpressionSyntax memberAccessExpression,
                ExpressionSyntax subExpression, 
                SyntaxToken previousToken)
            {
                if (previousToken != default(SyntaxToken))
                {
                    if (previousToken != memberAccessExpression.OperatorToken)
                    {
                        return SpecializedCollections.EmptyEnumerable<ITypeSymbol>();
                    }

                    // fall through
                }
                else
                {
                    if (subExpression != memberAccessExpression.Name)
                    {
                        // we're not on the name portion of hte member access expressoin.
                        // i.e. we're in "Foo" in "Foo.Bar".  We can't figure a name for this
                        // at all.
                        return SpecializedCollections.EmptyEnumerable<ITypeSymbol>();
                    }

                    // fall through
                }

                // We're right after the dot in "Foo.Bar".  The type for "Bar" should be
                // whatever type we'd infer for "Foo.Bar" itself.
                return InferTypes(memberAccessExpression);
            }
```

There should be no 'await' specialization code.

you can update the existin gcall to be:
```
(MemberAccessExpressionSyntax memberAccessExpression) => InferTypeInMemberAccessExpression(memberAccessExpression, expression, default(SyntaxToken)),
```

And your new call will be:

```
(MemberAccessExpressionSyntax memberAccessExpression) => InferTypeInMemberAccessExpression(memberAccessExpression, null, token),
```